### ISSUE TITLE
[GR-49355] System.console().readPassword broke in Oracle GraalVM 21 (compared to CE 22.3.0)

### ISSUE THREAD

**Issue by borkdude** [oracle/graal#7567]
Repro:

``` java
public class Repro {

    public static void main(String [] args) {
        System.console().readPassword();
    }

}
```

Compile and run:

```
$ ./repro
Exception in thread "main" java.lang.UnsatisfiedLinkError: jdk.internal.io.JdkConsoleImpl.echo(Z)Z [symbol: Java_jdk_internal_io_JdkConsoleImpl_echo or Java_jdk_internal_io_JdkConsoleImpl_echo__Z]
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:152)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:54)
	at java.base@21/jdk.internal.io.JdkConsoleImpl.echo(Native Method)
	at java.base@21/jdk.internal.io.JdkConsoleImpl.readPassword(JdkConsoleImpl.java:101)
	at java.base@21/jdk.internal.io.JdkConsoleImpl.readPassword(JdkConsoleImpl.java:158)
	at java.base@21/java.io.ProxyingConsole.readPassword(ProxyingConsole.java:128)
	at Repro.main(Repro.java:4)
	at java.base@21/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
```

**Comment by rbergmanaf** [comment]
We've run into this same issue trying to update from 22.3 as well.

**Comment by fniephaus** [comment]
This was fixed via https://github.com/oracle/graal/commit/8306a07394c23d24714855d6b6cf9e3758f23102 and will be backported.

**Comment by borkdude** [comment]
Upgrading to 21.0.1 didn't help

**Comment by fniephaus** [comment]
The fix didn't make it into the 21.0.1 patch update, but it will be included in the next one.

**Comment by michaeljmarshall** [comment]
I see that the planned release for 21.0.2 is in January 16, 2024 based on this page: https://docs.oracle.com/en/graalvm/enterprise/21/docs/release-calendar/#planned-relases. Do bugs ever trigger an earlier than planned patch release?
